core/state/statedb.go

```diff
@@ -604,9 +634,15 @@ func (s *StateDB) getDeletedStateObject(addr common.Address) *stateObject {
 			return nil
 		}
 	}
+
 	// Insert into the live set
 	obj := newObject(s, addr, data)
 	s.setStateObject(obj)
+	if s.witness != nil && s.snap != nil {
+		// when building witness with snap enabled, prefetch all read accounts to later be collected and
+		// included in the witness
+		s.readPrefetcher.prefetch(common.Hash{}, s.originalRoot, common.Address{}, [][]byte{addr[:]})
+	}
```

Comment:
This won't capture the proofs for empty account reads, those get short circut above.

Reply:
I can create a state test to try and trigger this.